include/revng/AutoEnforcer/AutoEnforcerContext.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <initializer_list>
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringMap.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/Casting.h"
+
+#include "revng/AutoEnforcer/KindsRegistry.h"
+#include "revng/AutoEnforcer/SavableObject.h"
+
+namespace AutoEnforcer {
+class NamedGlobal {
+public:
+  NamedGlobal(llvm::StringRef Name, SavableObjectBase &Obj) :
+    Name(Name), Obj(&Obj) {}
+
+  llvm::StringRef getName() const { return Name; }
+
+  SavableObjectBase &getObj() const { return *Obj; }
+
+private:
+  llvm::StringRef Name;
+  SavableObjectBase *Obj;
+};
+
+class Context {
+public:
+  explicit Context(KindsRegisty registry) : KRegistry(std::move(registry)) {}
+  Context();
+  Context(llvm::ArrayRef<NamedGlobal> Globals);
+  Context(const Context &) = delete;
+  Context(Context &&) = delete;
+  Context &operator=(const Context &) = delete;
+  Context &operator=(Context &&) = delete;
+  ~Context() = default;
+
+  template<typename T>
+  llvm::Expected<T *> get(llvm::StringRef Name) const {
```

Comment:
It's not obvious to me what `Context::get` should return. What about `getGlobal`? This is kind of related to my previous comment about the role of this class.